kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQuerySchemaConverter.java

```diff
@@ -106,7 +103,9 @@ public BigQuerySchemaConverter(boolean allFieldsNullable) {
           ConversionConnectException("Top-level Kafka Connect schema must be of type 'struct'");
     }
 
-    throwOnCycle(kafkaConnectSchema, new ArrayList<>());
+    if(throwOnCycle(kafkaConnectSchema, new ArrayList<>())) {
+      throw new ConversionConnectException("Kafka Connect schema contains cycle. See logs for detail");
```

Comment:
Is the "See logs for detail" bit necessary? Why not include information on the failing schema in the exception message itself?

This is an important distinction because the message of the exception that causes the connector to fail is made visible to users via the REST API via things like the `/connector/{connector}/status` endpoint, whereas logs are harder to access. We should try to include important information directly in exception messages when possible so that people don't have to look at logs to get the info they need. See https://github.com/confluentinc/kafka-connect-bigquery/pull/150 for an example of where we're trying to improve things on that front.

Reply:
Fair point. Done